### DISCUSSION TITLE
Difference from vscode-server

### DISCUSSION THREAD

**Discussion by jyboy** [coder/code-server#5336] [General] [answered]
Recently the VS Code Server was released.
https://code.visualstudio.com/blogs/2022/07/07/vscode-server

**Comment by karuboniru** [comment]
My experience is that coder/code-server can be proxied under a Nginx path while microsoft's can only run at root path now

**Comment by code-asher** [comment, marked as answer]
I saw a comment in my email that seems to have been deleted and figured we should answer this one since it seems we missed it.

At this point code-server and upstream's server are almost identical except code-server makes things more friendly for self-hosting.

For example things like the ability to host at sub-paths as mentioned, self-contained web views that do not call out to Microsoft's servers, the ability to add your own marketplace, collect your own telemetry, and [some other things](https://github.com/coder/code-server/tree/main/patches).

Some of these changes appear very unlikely to ever be adopted by Microsoft (marketplace, telemetry, allowing all extensions to use proposed API by default, some other changes probably) but some I expect to make their way upstream, further closing the gap.  But at the moment it looks like there will always be some subtle differences.
